### ISSUE TITLE
Define symbols using arithmetic on imported symbols

### ISSUE THREAD

**Issue by pinobatch** [gbdev/rgbds#1146]
I'm working on a local variable allocator that counts how many HRAM variables a subroutine uses so that its callers know how many HRAM variables to skip when placing their own locals. I had planned to build it as a macro pack where macros would do this:

1. Import the address of the end of local variables for each callee
2. Calculate their maximum, giving the address of the start of local variables for the present subroutine
3. Add the size of local variables for the present subroutine, giving the address of the end of local variables for the present subroutine
4. Export this address for use by the present subroutine's callers

I wrote an example of steps 1, 3, and 4 in ca65 ([locals-sample-ca65.s.txt](https://github.com/gbdev/rgbds/files/11837538/locals-sample-ca65.s.txt)), which assembled successfully. The right side of the symbol definition isn't constexpr, and [honey badger don't give a bleep](https://youtu.be/4r7wHMg5Yjg?t=28) because honey badger can put it in a patch and make the linker fix it up.

However, a line-by-line translation of this code to RGBASM syntax ([locals-sample-rgbasm.s.txt](https://github.com/gbdev/rgbds/files/11837545/locals-sample-rgbasm.s.txt)) produces an error at the `DEF ... EQU`.
```
error: locals-sample-rgbasm.s.txt(3):
    Expected constant expression: 'calleeFunc_hLocalsEnd' is not constant at assembly time
error: Assembly aborted (1 error)!
```

suggested labels: enhancement, ca65 parity

**Comment by aaaaaa123456789** [comment]
Can you please post the files as gists or anywhere else where they'd render on a browser? Right now, your links force a download. Thanks!

**Comment by ISSOtm** [comment]
This seems close to #201; I'm thinking of either allowing to define "soft labels" (symbols that are defined using `EQU` syntax but aren't constant), or we'd have to create a separate numeric symbol type that stores any RPN expression (which would be much more heavy to implement).

I asked Pino to make an issue for this so the use case is recorded (it seems it can be shimmed with `EQUS` in the meantime), but I don't plan to touch this until after #201 is implemented, and decide from there.

**Comment by pinobatch** [comment]
Gists:
* [ca65 example](https://gist.github.com/pinobatch/69ab4f1ae82ea1d7e6b532388514f4dc)
* [RGBASM example](https://gist.github.com/pinobatch/f04482c035017c1f01e711f5e93b45a4)

I had posted the files as a download for two reasons.

1. I expected maintainers to download the files, visually compare them on a desktop or laptop computer to gain confidence that their lines correspond, and use the RGBASM one as a test case for changes related to the issue. I wanted to __avoid mistakes while copying and pasting__ the gist into a text editor that may have hindered comparing them and testing with them.
2. When I post a gist, I want to include context for future readers who discover it. When posting a new issue, I __don't have an issue URL__.

Going forward, are gists better than downloads in general for examples of input to RGBASM?

**Comment by Rangi42** [comment]
I've sometimes wanted to define "soft labels" in RAM code. Currently we sometimes write this pattern:

```
LOAD "RAM Code", WRAM0
wFunction:
	...
	db $cd ; call
.callee:
	ds 2
	...
ENDL
```

Since neither of these are possible:

```
	db LOW(OPCODE(call 0))
	ds 2
```

```
	call .callee: 0
```

With soft labels, we could do something like this:

```
	DEF .callee EQUL @ + 1
	call 0
```

**Comment by aaaaaa123456789** [comment]
Isn't that what you'd use EQUS for?

**Comment by pinobatch** [comment]
`EQUS` cannot be `EXPORT`ed.

**Comment by Rangi42** [comment]
`EQUS` also doesn't appear in .sym files.

**Comment by Rangi42** [comment]
Re: "soft labels", [fasm](https://flatassembler.net/docs.php?article=fasmg_manual) has those:

> The "label" keyword followed by a symbol identifier is an alternative way of defining a label. In this basic form it is equivalent to a definition made with ":", but it occupies an entire line.

it also allows for attaching metadata to labels, which could be a solution for #1348:

> However with this command it is possible to provide more settings for the defined label. The identifier may be optionally followed by the ":" token and then an additional value to be associated with this label (usually denoting the size of the labeled entity). The assembler has a number of built-in constants defining various sizes for this purpose, but this value can also be provided as a plain number.
> 
> ```
>     label character:byte
>     label char:1
> ```

**Comment by Rangi42** [comment]
If we had user-defined functions (#201), zero-arg functions would serve this role... *if* they were exportable. That probably wouldn't be an initial MVP feature of user-defined functions. But eventually allowing them to be exported would be simpler than allowing `EQU` constants to be non-constant undetermined values.
